sidecar/src/service/runtime_info.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO-EK: Re-evaluate access levels for the structs and methods in this file after refactor,
+// before merging
+// TODO-EK: Better understand why apps and actions are conflated in this struct before merging. And,
+// if they should be separated.
```

Comment:
Because service and environment names are only known late. Hence we first queue actions - once the service and env names are finalized, we create the app and flush everything from queue to app.

Reply:
Thanks, that's good to know. I'll update the comment to reflect this.